classes/Cart.php

```diff
@@ -2415,6 +2415,10 @@ public function getPackageList($flush = false)
                 }
             }
 
+            if (!$product['is_virtual'] && empty($product['carrier_list'])) {
+                return array();
+            }
```

Comment:
It is sufficient that one product does not have the carriers available to block the ordering process. This will display the usual message: ```No carrier available for this delivery address```. Predict the possibility of sending a message for this case like: ```No carrier found for the PRODUCT X, please update your cart``` ?!